src/VaultSharp/V1/AuthMethods/Token/TokenAuthMethodProvider.cs

```diff
@@ -23,14 +23,14 @@ public async Task<Secret<object>> CreateTokenAsync(CreateTokenRequest createToke
 
             var suffix = "create";
 
-            if (createTokenRequest.NoParent)
+            if (request.CreateOrphan)
             {
                 suffix = "create-orphan";
             }
 
-            if (!string.IsNullOrWhiteSpace(createTokenRequest.RoleName))
+            if (!string.IsNullOrWhiteSpace(request.RoleName))
```

Comment:
i think role is applicable for non-orphan tokens. So can you please add the check for orphan and only when not orphan, then do this role suffix?

otherwise, somebody calling with CreateOrphan=true and roleName="abc" will end up needing sudo access and a non-orphan token being created.

Reply:
actually, i'll add that check during version tick